superset/viz.py

```diff
@@ -290,9 +285,9 @@ def get_payload(self, query_obj=None):
         payload = self.get_df_payload(query_obj)
 
         df = payload.get('df')
-        if df is not None and len(df.index) == 0:
-            raise Exception('No data')
         if self.status != utils.QueryStatus.FAILED:
+            if df is None or df.empty:
+                raise Exception('No data')
```

Comment:
Note I'm not certain why we need to throw an exception here, as previously the error would be present in `error_message`.